src/Database/LSMTree/Internal/BlobFile.hs

```diff
@@ -51,24 +52,35 @@ instance NFData BlobSpan where
 
 -- | Open the given file to make a 'BlobFile'. The finaliser will close and
 -- delete the file.
+--
+-- REF: the resulting reference must be released once it is no longer used.
+--
+-- ASYNC: this should be called with asynchronous exceptions masked.
```

Comment:
Might be good to clarify that asynchronous exceptions should be masked by the caller simply because this allocates a resource.